server/aws-lsp-codewhisperer/src/language-server/utils.ts

```diff
@@ -131,3 +133,15 @@ export const getUserAgent = (initializeParams: InitializeParams, serverInfo?: Se
 
     return items.join(' ')
 }
+
+export function getLoginTypeFromProvider(credentialsProvider: CredentialsProvider): LoginType {
+    if (credentialsProvider.hasCredentials('iam')) {
+        return 'iam'
+    }
+    const connectionMetadata = credentialsProvider.getConnectionMetadata()
+    return connectionMetadata && connectionMetadata.sso && connectionMetadata.sso.startUrl
```

Comment:
nit: this conditions chain can be replaced with optional chaining `connectionMetadata?.sso?.startUrl?.includes(builderIdStartUrl)`